app/letter_attachment/rest.py

```diff
@@ -41,3 +43,24 @@ def create_letter_attachment():
     dao_update_template(template)
 
     return jsonify(letter_attachment.serialize()), 201
+
+
+@letter_attachment_blueprint.route("/letter-attachment/<uuid:letter_attachment_id>/archive", methods=["POST"])
+def archive_letter_attachment(letter_attachment_id):
+    data = request.get_json()
+
+    validate(data, post_archive_letter_attachment_schema)
+
+    letter_attachment = LetterAttachment.query.get_or_404(letter_attachment_id)
+    # I found I had to do this to set up the versioning properly, rather than use the template directly.
+    if (
+        letter_attachment.template is None
+        or (template := dao_get_template_by_id(letter_attachment.template.id)) is None
+    ):
+        raise InvalidRequest("template-does-not-exist", 400)
```

Comment:
```suggestion
    letter_attachment = LetterAttachment.query.get_or_404(letter_attachment_id)
    if not (template := letter_attachment.template):
        raise InvalidRequest("template-does-not-exist", 400)
```

I think we should be able to do this? But not sure exactly what problem you ran into with the comment as I'm not seeing it. 🤔

Reply:
It occurred when I assigned template before the if statement. Your version works fine. I don't know the intricacies of the template versioning system, I might dig if I get a chance.

Reply:
Unfortunately we have about 3 versioning systems.. :(